Storage/src/Connection/Rest.php

```diff
@@ -98,6 +110,7 @@ public function __construct(array $config = [])
         ));
 
         $this->projectId = $this->pluck('projectId', $config, false);
+        $this->restRetryFunction = (isset($config['restRetryFunction'])) ? $config['restRetryFunction'] : null;
```

Comment:
Why do we need this? I do not think there's a real usecase for a user-suppled value for `restRetryFunction`

Reply:
It was never there in the constructor, but the users could have always passed a `restRetryFunction` to their need and it is also used in the `RequestWrapper::send`.

The only extra thing we have done is document it.

We have recommended people to use it(as a config var, not in constructor) in other issues too.
#5230 #4062